--- v8-5.6.331/src/v8.gyp.omv~	2016-12-16 12:00:25.573755836 +0100
+++ v8-5.6.331/src/v8.gyp	2016-12-16 12:00:55.198041390 +0100
@@ -1872,6 +1872,11 @@
               'USING_V8_BASE_SHARED',
             ],
           },
+          'conditions': [
+            ['soname_version!=""', {
+              'product_extension': 'so.<(soname_version)',
+            }],
+          ],
         }],
         ['OS=="linux"', {
             'link_settings': {
@@ -2121,6 +2126,11 @@
             'defines': [ 'USING_V8_PLATFORM_SHARED' ],
           },
           'defines': [ 'BUILDING_V8_PLATFORM_SHARED' ],
+          'conditions': [
+            ['soname_version!=""', {
+              'product_extension': 'so.<(soname_version)',
+            }],
+          ],
         }]
       ],
       'direct_dependent_settings': {
